File: samples/maui/KsDialogs.Sample.Maui/InlineToastViewModel.cs
namespace KsDialogs.Sample.Maui;

/// <summary>
/// インライン経路のカスタム Toast の ViewModel。
/// </summary>
/// <remarks>中身はこの場で渡すため、この型はレジストリに登録しない (core/ADR-0013)。</remarks>
/// <param name="message">Toast に表示するメッセージ。</param>
public sealed class InlineToastViewModel(string message) : IToastViewModel
{
    /// <summary>Toast に表示するメッセージ。</summary>
    public string Message { get; } = message;
}

File: samples/kmp/iosApp/KsDialogsSampleKmp/InlineToastViewModel.swift
import KsDialogs

/// インライン経路のカスタム Toast の ViewModel。
///
/// 中身をその場で渡す表示は各 OS の Native API にしかないため、この ViewModel も OS 側に置く。
/// この型はレジストリに登録しない (core/ADR-0013)。
final class InlineToastViewModel: ToastViewModel {
    /// Toast に表示するメッセージ。
    let message: String

    init(message: String) {
        self.message = message
    }
}
